src/c++/perf_analyzer/test_load_manager_base.h

```diff
@@ -142,6 +169,42 @@ class TestLoadManagerBase {
   }
   std::shared_ptr<cb::MockClientStats> GetStats() { return stats_; }
   void ResetStats() { stats_->Reset(); }
+
+  static PerfAnalyzerParameters GetSequenceTestParamsHelper(bool is_async)
```

Comment:
Not a huuuuge fan of how nested these test functions are getting especially since they're using macros that are really only shown in the doctest docs as being used directly in `TEST_CASE`es. I see the value of having complex subcases, but I wish there was a way to keep it all in a `TEST_CASE`. Maybe it's an unnecessary preference.

Reply:
they don't provide an easy way to get a cross product of a bunch of cases. Nested functions prevents a ton of cutting and pasting.

They do have examples of using subcases in functions [here](https://github.com/doctest/doctest/blob/7b9885133108ae301ddd16e2651320f54cafeba7/examples/all_features/subcases.cpp#L49)

Reply:
Fair enough 👍